agent/agent.go

```diff
@@ -277,6 +127,30 @@ func (a *Agent) ResyncBackupList() {
 	}
 	err = lock.Release()
 	if err != nil {
-		log.Printf("[ERROR] backup: unable to release backup lock for %v:%v\n", lock, err)
+		a.log.Error(pbm.CmdResyncBackupList, "", "reslase lock %v: %v", lock, err)
+	}
+}
+
+func (a *Agent) aquireLock(l *pbm.Lock, m func(name string) error) (got bool, err error) {
+	got, err = l.Acquire()
+	if err == nil {
+		return got, nil
+	}
+
+	switch err.(type) {
+	case pbm.ErrConcurrentOp:
+		a.log.Info(pbm.CmdUndefined, "", "acquiring lock: %v", err)
+		return false, nil
+	case pbm.ErrWasStaleLock:
+		if m != nil {
+			name := err.(pbm.ErrWasStaleLock).Lock.BackupName
+			merr := m(name)
+			if merr != nil {
+				a.log.Warning(pbm.CmdUndefined, "", "failed to mark stale backup '%s' as failed: %v", name, merr)
+			}
+		}
+		return l.Acquire()
+	default:

```

Comment:
Is it's worth to log error as in `pbm.ErrConcurrentOp` ?

Reply:
It's gonna be logged by the caller. In the pbm.ErrConcurrentOp branch we're logging an info event since a returned error is nil.

Reply:
got it